qiskit/transpiler/passes/mapping/barrier_before_final_measurements.py

```diff
@@ -74,13 +74,14 @@ def run(self, dag):
         our_descendants = barrier_layer.descendants(new_barrier_node)
         our_qubits = final_qubits
```

Comment:
I have introduced a new pass called `MergeAdjacentBarriers` which takes a circuit and merges any barriers that can be merged into one. I have removed this functionality from the `BarrierBeforeFinalMeasurements` pass, so this pass now simply adds the barrier in, but it does not attempt to merge it with any already existing barriers.  Currently, so that the functionality of the `BarrierBeforeFinalMeasurements` is kept constant, this pass calls the `MergeAdjacentBarriers` pass so that the output is the same, but I think this should be removed in future.

Reply:
Yeah I also think we should not manually call the `MergeAdjacentBarriers` pass towards the end of `BarrierBeforeFinalMeasurements`, so that the passes can be standalone, which was the point of splitting them. Not blocking for this PR.